rasa/nlu/server.py

```diff
@@ -31,324 +52,276 @@ def create_argument_parser():
+        if "query" in request_params:
+            request_params["q"] = request_params.pop("query")
+        if "q" not in request_params:
+            request_params["q"] = ""
+        return await parse_response(request_params)
+
+    @app.post("/parse")
+    @requires_auth(app, token)
+    async def parse(request):
+        request_params = request.json
 
         if "query" in request_params:
             request_params["q"] = request_params.pop("query")
 
         if "q" not in request_params:
-            request.setResponseCode(404)
-            dumped = json_to_string({"error": "Invalid parse parameter specified"})
-            returnValue(dumped)
+            return response.json(
+                {"error": "Invalid parse parameter specified"}, status=404
+            )
         else:
-            data = self.data_router.extract(request_params)
-            try:
-                request.setResponseCode(200)
-                response = yield (
-                    self.data_router.parse(data)
-                    if self._testing
-                    else threads.deferToThread(self.data_router.parse, data)
-                )
-                returnValue(json_to_string(response))
-            except InvalidProjectError as e:
-                request.setResponseCode(404)
-                returnValue(json_to_string({"error": "{}".format(e)}))
-            except Exception as e:
-                request.setResponseCode(500)
-                logger.exception(e)
-                returnValue(json_to_string({"error": "{}".format(e)}))
-
-    @app.route("/version", methods=["GET", "OPTIONS"])
-    @requires_auth
-    @check_cors
-    def version(self, request):
+            return await parse_response(request_params)
+
+    @app.get("/version")
+    @requires_auth(app, token)
+    async def version(request):
         """Returns the Rasa server's version"""
 
-        request.setHeader("Content-Type", "application/json")
-        return json_to_string(
+        return response.json(
             {
                 "version": rasa.__version__,
                 "minimum_compatible_version": MINIMUM_COMPATIBLE_VERSION,
             }
         )
 
-    @app.route("/status", methods=["GET", "OPTIONS"])
-    @requires_auth
-    @check_cors
-    def status(self, request):
-        request.setHeader("Content-Type", "application/json")
-        return json_to_string(self.data_router.get_status())
+    @app.get("/status")
+    @requires_auth(app, token)
+    async def status(request):
+        return response.json(data_router.get_status())
 
-    def extract_json(self, content):
+    def extract_json(content):
         # test if json has config structure
         json_config = simplejson.loads(content).get("data")
 
         # if it does then this results in correct format.
         if json_config:
-
-            model_config = simplejson.loads(content)
-            data = json_config
+            return simplejson.loads(content), json_config
```

Comment:
I think we don't have to use simplejsons anymore, right? I guess we had this for Python 2? We should also check the requirements / setup.py